server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/CreateExternalApprovalTask.java

```diff
@@ -0,0 +1,99 @@
+import com.oceanbase.odc.service.flow.task.model.RuntimeTaskConstants;
+import com.oceanbase.odc.service.flow.util.FlowTaskUtil;
+import com.oceanbase.odc.service.integration.IntegrationService;
+import com.oceanbase.odc.service.integration.client.ApprovalClient;
+import com.oceanbase.odc.service.integration.model.ApprovalProperties;
+import com.oceanbase.odc.service.integration.model.IntegrationConfig;
+import com.oceanbase.odc.service.integration.model.TemplateVariables;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author gaoda.xy
+ * @date 2023/9/1 14:00
+ */
+@Slf4j
+public class CreateExternalApprovalTask extends BaseFlowableDelegate {
+
+    @Autowired
+    private FlowableAdaptor flowableAdaptor;
+    @Autowired
+    private IntegrationService integrationService;
+    @Autowired
+    private ApprovalClient approvalClient;
+    private final RetryExecutor retryExecutor = RetryExecutor.builder().retryIntervalMillis(1000).retryTimes(3).build();
+
+    @Override
+    protected void run(DelegateExecution execution) throws Exception {
+        String activityId = execution.getCurrentActivityId();
+        String processDefinitionId = execution.getProcessDefinitionId();
+        Optional<Optional<Long>> flowInstanceIdOpt = retryExecutor.run(
+                () -> flowableAdaptor.getFlowInstanceIdByProcessDefinitionId(processDefinitionId), Optional::isPresent);
+        if (!flowInstanceIdOpt.isPresent() || !flowInstanceIdOpt.get().isPresent()) {
+            log.warn("Flow instance id does not exist, activityId={}, processDefinitionId={}", activityId,
+                    processDefinitionId);
+            throw new IllegalStateException(
+                    "Can not find flow instance id by process definition id " + processDefinitionId);
+        }
+        Long flowInstanceId = flowInstanceIdOpt.get().get();
+        Optional<FlowApprovalInstance> instanceOpt =
+                flowableAdaptor.getApprovalInstanceByActivityId(activityId, flowInstanceId);
+        if (!instanceOpt.isPresent()) {
+            log.warn("Flow approval instance does not exist, activityId={}, flowInstanceId={}", activityId,
+                    flowInstanceId);
+            throw new IllegalStateException("Can not find instance by activityId " + activityId);
+        }
+        FlowApprovalInstance flowApprovalInstance = instanceOpt.get();
+        Long externalApprovalId = flowApprovalInstance.getExternalApprovalId();
+        if (Objects.nonNull(externalApprovalId)) {
+            log.info("Start create external approval instance, flowApprovalInstanceId={}, externalApprovalId={}",
+                    flowApprovalInstance.getId(), externalApprovalId);
+            IntegrationConfig config = integrationService.detailWithoutPermissionCheck(externalApprovalId);
+            ApprovalProperties properties = ApprovalProperties.from(config);
+            TemplateVariables variables = FlowTaskUtil.getTemplateVariables(execution.getVariables());
```

Comment:
don't put `TemplateVariables` into flow variables. `TemplateVariables` contains `Map<String, Object>`, which may lead to error~

Reply:
But, the `TemplateVariables` is needed by approval integration, and used to build HTTP body.
In preview testing, it works as expected. So can we keep it as before?